gui/src/main.rs

```diff
@@ -43,68 +58,156 @@ fn run() -> Result<(), ApplicationError> {
             .unwrap();
     }
 
-    let args = CliArgs::try_parse().map_err(ApplicationError::ParseArgs)?;
+    // TODO: check if already configured and skip wizard
+    run_wizard().map_err(ApplicationError::RunWizard)?;
 
     if let Some(debug_addr) = args.debug_addr() {
         let debug_server = DebugServer::new(debug_addr)
             .map_err(|e| ApplicationError::StartDebugServer(e, debug_addr))?;
 
-        let debug_client = debug_server
+        let _debug_client = debug_server
             .accept()
             .map_err(ApplicationError::CreateDebugClient)?;
     }
 
-    let app = App::new()?;
+    run_main_app()?;
 
-    app.run()?;
+    Ok(())
+}
+
+fn run_main_app() -> Result<(), ApplicationError> {
+    let main_window = ui::MainWindow::new().map_err(ApplicationError::CreateMainWindow)?;
+
+    let graphics_api = graphics::DefaultApi::new().map_err(ApplicationError::InitGraphicsApi)?;
+
+    let devices: Vec<SharedString> = graphics_api
+        .physical_devices()
+        .into_iter()
+        .map(|d| SharedString::from(d.name()))
+        .collect();
+
+    main_window.set_devices(ModelRc::new(VecModel::from(devices)));
+
+    let profiles = ModelRc::new(
+        VecModel::from(vec![profile::Profile::default()])
+            .map(|p| SharedString::from(String::from(p.name().to_string_lossy()))),
+    );
+
+    main_window.set_profiles(profiles.clone());
+
+    main_window.on_start_game(|_index| {
+        let screen = ui::Screen::new().unwrap();
+
+        screen.show().unwrap();
```

Comment:
Is it possible to reuse the same Windows like how Qt does?

Reply:
It might be. Let's put a todo there for since there's a lot of work to be done there anyway